src/objective/rank_xendcg_objective.hpp

```diff
@@ -0,0 +1,141 @@
+#ifndef LIGHTGBM_OBJECTIVE_RANK_XENDCG_OBJECTIVE_HPP_
```

Comment:
Can you please update https://github.com/microsoft/LightGBM/tree/master/windows and include the new file there?

Reply:
Done.